application/models/Survey.php

```diff
@@ -2069,4 +2068,22 @@ public function setToInherit()
         }
     }
 
+    /**
+     * Find all public surveys
+     * @return Survey[]
+     */
+    public function findAllPublic()
+    {
+        $oCriteria = new CDbCriteria();
+        $oCriteria->condition = "listpublic = 'Y' or listpublic = 'I'";
```

Comment:
Also check if it's active?

Reply:
That's aleady included on the scopes (filters) before on the chain, which are taken into account by doing `$this->findAll()`.

Reply:
Are you sure...?

Reply:
Yes. I mean, we are changing the `findAll()` by a `findAllPublic()`, which calls internally `findAll` with a custom criteria.

Goging deeper: `findAll` calls `query()`, which in turns calls `applyScopes`, which applies prior scopes to the latest criteria. https://www.yiiframework.com/doc/api/1.1/CActiveRecord#query-detail

You can see debug in picture attached.
![image](https://user-images.githubusercontent.com/4942495/90770705-8f13d680-e2c8-11ea-8de7-b1d7342d9fdc.png)
From the watch: `$c->condition = "(active = 'Y') AND ((startdate <= :now1 OR startdate IS NULL) AND (expires >= :now2 OR expires IS NULL))"`

Reply:
Oh God, so much "spooky action at a distance". Thanks for having good insight!